core/build.gradle

```diff
@@ -53,7 +63,47 @@ animalsniffer {
 
 import net.ltgt.gradle.errorprone.CheckSeverity
 
+def replaceBytes(byte[] haystack, byte[] needle, byte[] replacement) {
+    int i = Bytes.indexOf(haystack, needle);
+    assert i != -1;
+    byte[] result = new byte[haystack.length - needle.length + replacement.length];
+    System.arraycopy(haystack, 0, result, 0, i);
+    System.arraycopy(replacement, 0, result, i, replacement.length);
+    System.arraycopy(haystack, i + needle.length, result, i + replacement.length, haystack.length - i - needle.length);
+    return result;
+}
+
+def bigEndian(int value) {
+    return [value >> 8, value & 0xFF] as byte[];
+}
+
+def replaceConstant(File file, String needle, String replacement) {
+    // CONSTANT_Utf8_info. https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.4.7
+    byte[] needleBytes = Bytes.concat(
+        [1] as byte[], bigEndian(needle.length()), needle.getBytes(US_ASCII));
+    byte[] replacementBytes = Bytes.concat(
+        [1] as byte[], bigEndian(replacement.length()), replacement.getBytes(US_ASCII));
+    file.setBytes(replaceBytes(file.getBytes(), needleBytes, replacementBytes));
+}
+
 plugins.withId("java") {
+    compileJava {
```

Comment:
Would it make sense to add a comment explaining the reason why we do this? Or, perhaps, a tracking ticket to capture the context, and the plan detailing when we're planning to remove this.